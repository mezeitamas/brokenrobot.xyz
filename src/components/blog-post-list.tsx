import React from 'react';
import type { FunctionComponent, ReactElement } from 'react';

import { BlogPostListItem } from './blog-post-list-item';

type BlogPostListProps = {
    posts: { title: string; excerpt: string; published: string; slug: string }[];
};

const BlogPostList: FunctionComponent<BlogPostListProps> = ({ posts }): ReactElement => {
    return (
        <div>
            {posts.map(({ title, excerpt, published, slug }) => (
                <BlogPostListItem
                    key={slug}
                    title={title}
                    excerpt={excerpt}
                    published={published}
                    slug={slug}
                />
            ))}
        </div>
    );
};

export { BlogPostList };
