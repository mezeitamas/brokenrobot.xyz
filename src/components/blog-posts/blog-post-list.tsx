import React from 'react';
import type { FunctionComponent, ReactElement } from 'react';

import { BlogPostListItem } from './blog-post-list-item';

type BlogPostListProps = {
    posts: { title: string; excerpt: string; published: string; publishedFormatted: string; slug: string }[];
    HeadingType: keyof JSX.IntrinsicElements;
};

const BlogPostList: FunctionComponent<BlogPostListProps> = ({ posts, HeadingType }): ReactElement => {
    return (
        <div>
            {posts.map(({ title, excerpt, published, publishedFormatted, slug }) => (
                <BlogPostListItem
                    key={slug}
                    title={title}
                    excerpt={excerpt}
                    published={published}
                    publishedFormatted={publishedFormatted}
                    slug={slug}
                    HeadingType={HeadingType}
                />
            ))}
        </div>
    );
};

export { BlogPostList };

export type { BlogPostListProps };
