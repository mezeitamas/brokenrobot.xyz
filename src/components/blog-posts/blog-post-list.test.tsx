import React from 'react';

import { create } from 'react-test-renderer';

import { BlogPostList } from './blog-post-list';
import type { BlogPostListProps } from './blog-post-list';

describe('<BlogPostList />', () => {
    test('should render with h2 heading', () => {
        // Given
        const props: BlogPostListProps = {
            posts: [
                {
                    title: 'Advanced static website hosting with Amazon S3 and CloudFront',
                    excerpt:
                        'Today we will dive deep into the world of hosting a static website on Amazon S3 using CloudFront. If you’re looking for a secure, reliable, scalable, cost effective and performant solution, you’ve come to the right place. In this blog post, we’ll...',
                    published: '2023-05-29T18:00:00.000Z',
                    publishedFormatted: 'May 29, 2023',
                    slug: 'advanced-static-website-hosting-with-amazon-s3-and-cloudfront'
                },
                {
                    title: 'Hosting a static website on Amazon S3',
                    excerpt:
                        'Hosting a simple static website on Amazon Web Services (AWS) can be daunting at first, but luckily it’s quite straightforward. One option for hosting a static website on AWS is to use Amazon S3. In this post, we will walk through the steps to host a...',
                    published: '2023-05-04T17:58:32.000Z',
                    publishedFormatted: 'May 4, 2023',
                    slug: 'hosting-a-static-website-on-amazon-s3'
                },
                {
                    title: 'Hello, World!',
                    excerpt:
                        'As a software engineer, I can’t think of a better way to start my blog than with the classic “Hello, World!“. For those unfamiliar with the term, “Hello, World!” is often the first program developers write when learning a new programming language or...',
                    published: '2023-05-01T10:10:05.000Z',
                    publishedFormatted: 'May 1, 2023',
                    slug: 'hello-world'
                }
            ],
            HeadingType: 'h2'
        };

        // When
        const component = create(<BlogPostList {...props} />).toJSON();

        // Then
        expect(component).toMatchSnapshot();
    });

    test('should render with h3 heading', () => {
        // Given
        const props: BlogPostListProps = {
            posts: [
                {
                    title: 'Advanced static website hosting with Amazon S3 and CloudFront',
                    excerpt:
                        'Today we will dive deep into the world of hosting a static website on Amazon S3 using CloudFront. If you’re looking for a secure, reliable, scalable, cost effective and performant solution, you’ve come to the right place. In this blog post, we’ll...',
                    published: '2023-05-29T18:00:00.000Z',
                    publishedFormatted: 'May 29, 2023',
                    slug: 'advanced-static-website-hosting-with-amazon-s3-and-cloudfront'
                },
                {
                    title: 'Hosting a static website on Amazon S3',
                    excerpt:
                        'Hosting a simple static website on Amazon Web Services (AWS) can be daunting at first, but luckily it’s quite straightforward. One option for hosting a static website on AWS is to use Amazon S3. In this post, we will walk through the steps to host a...',
                    published: '2023-05-04T17:58:32.000Z',
                    publishedFormatted: 'May 4, 2023',
                    slug: 'hosting-a-static-website-on-amazon-s3'
                },
                {
                    title: 'Hello, World!',
                    excerpt:
                        'As a software engineer, I can’t think of a better way to start my blog than with the classic “Hello, World!“. For those unfamiliar with the term, “Hello, World!” is often the first program developers write when learning a new programming language or...',
                    published: '2023-05-01T10:10:05.000Z',
                    publishedFormatted: 'May 1, 2023',
                    slug: 'hello-world'
                }
            ],
            HeadingType: 'h3'
        };

        // When
        const component = create(<BlogPostList {...props} />).toJSON();

        // Then
        expect(component).toMatchSnapshot();
    });
});
