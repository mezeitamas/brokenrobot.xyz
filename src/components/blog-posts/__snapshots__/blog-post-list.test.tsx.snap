// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<BlogPostList /> should render with h2 heading 1`] = `
<div>
  <article
    className="flex flex-col items-start justify-between"
  >
    <header>
      <h2>
        <span
          className="hover:underline"
        >
          <a
            href="/blog/advanced-static-website-hosting-with-amazon-s3-and-cloudfront/"
          >
            Advanced static website hosting with Amazon S3 and CloudFront
          </a>
        </span>
      </h2>
      <time
        dateTime="2023-05-29T18:00:00.000Z"
      >
        May 29, 2023
      </time>
      <p>
        Today we will dive deep into the world of hosting a static website on Amazon S3 using CloudFront. If you’re looking for a secure, reliable, scalable, cost effective and performant solution, you’ve come to the right place. In this blog post, we’ll...
      </p>
    </header>
  </article>
  <article
    className="flex flex-col items-start justify-between"
  >
    <header>
      <h2>
        <span
          className="hover:underline"
        >
          <a
            href="/blog/hosting-a-static-website-on-amazon-s3/"
          >
            Hosting a static website on Amazon S3
          </a>
        </span>
      </h2>
      <time
        dateTime="2023-05-04T17:58:32.000Z"
      >
        May 4, 2023
      </time>
      <p>
        Hosting a simple static website on Amazon Web Services (AWS) can be daunting at first, but luckily it’s quite straightforward. One option for hosting a static website on AWS is to use Amazon S3. In this post, we will walk through the steps to host a...
      </p>
    </header>
  </article>
  <article
    className="flex flex-col items-start justify-between"
  >
    <header>
      <h2>
        <span
          className="hover:underline"
        >
          <a
            href="/blog/hello-world/"
          >
            Hello, World!
          </a>
        </span>
      </h2>
      <time
        dateTime="2023-05-01T10:10:05.000Z"
      >
        May 1, 2023
      </time>
      <p>
        As a software engineer, I can’t think of a better way to start my blog than with the classic “Hello, World!“. For those unfamiliar with the term, “Hello, World!” is often the first program developers write when learning a new programming language or...
      </p>
    </header>
  </article>
</div>
`;

exports[`<BlogPostList /> should render with h3 heading 1`] = `
<div>
  <article
    className="flex flex-col items-start justify-between"
  >
    <header>
      <h3>
        <span
          className="hover:underline"
        >
          <a
            href="/blog/advanced-static-website-hosting-with-amazon-s3-and-cloudfront/"
          >
            Advanced static website hosting with Amazon S3 and CloudFront
          </a>
        </span>
      </h3>
      <time
        dateTime="2023-05-29T18:00:00.000Z"
      >
        May 29, 2023
      </time>
      <p>
        Today we will dive deep into the world of hosting a static website on Amazon S3 using CloudFront. If you’re looking for a secure, reliable, scalable, cost effective and performant solution, you’ve come to the right place. In this blog post, we’ll...
      </p>
    </header>
  </article>
  <article
    className="flex flex-col items-start justify-between"
  >
    <header>
      <h3>
        <span
          className="hover:underline"
        >
          <a
            href="/blog/hosting-a-static-website-on-amazon-s3/"
          >
            Hosting a static website on Amazon S3
          </a>
        </span>
      </h3>
      <time
        dateTime="2023-05-04T17:58:32.000Z"
      >
        May 4, 2023
      </time>
      <p>
        Hosting a simple static website on Amazon Web Services (AWS) can be daunting at first, but luckily it’s quite straightforward. One option for hosting a static website on AWS is to use Amazon S3. In this post, we will walk through the steps to host a...
      </p>
    </header>
  </article>
  <article
    className="flex flex-col items-start justify-between"
  >
    <header>
      <h3>
        <span
          className="hover:underline"
        >
          <a
            href="/blog/hello-world/"
          >
            Hello, World!
          </a>
        </span>
      </h3>
      <time
        dateTime="2023-05-01T10:10:05.000Z"
      >
        May 1, 2023
      </time>
      <p>
        As a software engineer, I can’t think of a better way to start my blog than with the classic “Hello, World!“. For those unfamiliar with the term, “Hello, World!” is often the first program developers write when learning a new programming language or...
      </p>
    </header>
  </article>
</div>
`;
